### PR TITLE
add argument to the detect

### PR DESCRIPTION
**Pull request by rshtg**
Very needed isomorphic browser detection

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -153,11 +153,15 @@ const operatingSystemRules: OperatingSystemRule[] = [
   ['Search Bot', SEARCHBOT_OS_REGEX],
 ];
 
-export function detect(): BrowserInfo | BotInfo | NodeInfo | null {
+export function detect(userAgent: string | null = null): BrowserInfo | BotInfo | NodeInfo | null {
```
Comment (DamonOehlman) [inline src/index.ts]: How do you feel about using the `?:` argument form instead to have the `userAgent` argument receive an `undefined` value if not specified?  Personally, I feel like this is probably more in line with what happens when someone calls the function with zero args.

e.g.

```ts
detect(userAgent?: string): ...
```

File: src/index.ts [hunk 2]
```diff
@@ -153,11 +153,15 @@ const operatingSystemRules: OperatingSystemRule[] = [
...
   if (typeof navigator !== 'undefined') {
     return parseUserAgent(navigator.userAgent);
   }
 
+  if (userAgent) {
```
Comment (DamonOehlman) [inline src/index.ts]: I think we should push this above the `navigator` check. I think if someone has gone to the effort of specifying the `userAgent` argument that should take priority over everything else.

Additionally, I think being specific in the `if` clause is probably worthwhile, e.g.:

```ts
if (userAgent !== undefined) {
  ...
}
```

Comment (rshtg) [reply src/index.ts]: Yeah, think it's better to place the condition before the `navigator` check. Thanks for an advice :)

**Comment by DamonOehlman** [comment]
Published as [`4.6.0`](https://github.com/DamonOehlman/detect-browser/releases/tag/v4.6.0)

**Pull request merged**